packages/loader/container-loader/src/naming.md

```diff
@@ -1,7 +1,7 @@
 * `IContainer` (public):
   * `connectionState: ConnectionState`
     * `Disconnected`: no socket connection to delta server
-    * `Connecting`: socket connection established but we're not caught up
+    * `Pending`: socket connection established but we're not caught up
       * Unable to send outbound messages
```

Comment:
@vladsud This is from the comments in loader.ts in container-definitions, but doesn't look accurate from the code.  Outbound queue is very simple, and managed by `ConnectionManager` - as soon as `connect` event fires from `DeltaManager` that queue is resumed and if someone submitted a message it would fly.

So where does this notion come from?  Is there some hidden expectation that the runtime won't start submitting ops until `connected` event has fired or something...?  Didn't look into it just guessing.

Reply:
Runtime does not know about "connect" event, so it looks to runtime as if we are disconnected, and thus it will not send ops. Container.submitMessage() checks for violations.

I'd say that pausing of queues is orthogonal to management of connection states, or rather it should be, but clearly, it's more complicated than that. Might be worth cleaning it a bit.